tests/test_load.py

```diff
@@ -231,6 +245,21 @@ def test_LocalDatasetModuleFactoryWithoutScript(self):
         assert importlib.import_module(module_factory_result.module_path) is not None
         assert os.path.isdir(module_factory_result.builder_kwargs["base_path"])
 
+    def test_LocalDatasetModuleFactoryWithoutScript_with_data_dir(self):
+        factory = LocalDatasetModuleFactoryWithoutScript(self._data_dir, data_dir=self._sub_data_dir)
```

Comment:
I think a more probable case is when data_dir is a subdirectory of the base path:
```python
LocalDatasetModuleFactoryWithoutScript(data_dir_with_two_levels, data_dir="subdir1")
```

where data_dir_with_two_levels is like
```
data_dir2/
  subdir1/
    train.json
    test.json
  subdir2
    train.json
    test.json
```

We would also need the same test for `HubDatasetModuleFactoryWithoutScript`

Reply:
[23920a2](https://github.com/huggingface/datasets/pull/4144/commits/23920a290f53ba6845cc2bf880c00b8b9a09f0f5)
i've modified the test for `LocalDatasetModuleFactoryWithoutScript`

for `HubDatasetModuleFactoryWithoutScript` i think it's enough just to change checking if num files in each split is == 1 instead of > 0, i.e. only files from `data2` belong to dataset, we know in advance that there are one file per split (see dir structure in a [repo](https://huggingface.co/datasets/mariosasko/test_multi_dir_dataset/tree/main)